org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/util/TypeFilter.java

```diff
@@ -70,16 +76,52 @@ public static boolean isFiltered(TypeNameMatch match) {
 		}
 	}
 
+	/**
+	 * Remove the type filter if any of the imported element matches.
+	 */
+	public synchronized void removeFilterIfMatched(Collection<String> importedElements) {
+		if (importedElements == null || importedElements.isEmpty()) {
+			return;
+		}
+
+		StringMatcher[] matchers = getStringMatchers();
+		this.fStringMatchers = Arrays.stream(matchers).filter(m -> {
+			for (String importedElement : importedElements) {
+				if (m.match(importedElement)) {
+					return false;
+				}
+			}
+			return true;
+		}).toArray(size -> new StringMatcher[size]);
+	}
+
+
 	private StringMatcher[] fStringMatchers;
 
 	public TypeFilter() {
 		fStringMatchers= null;
-		PreferenceConstants.getPreferenceStore().addPropertyChangeListener(this);
+		getPreferenceStore().addPreferenceChangeListener(this);
+	}
+
+	private IEclipsePreferences getPreferenceStore() {
+		return InstanceScope.INSTANCE.getNode(JavaManipulation.getPreferenceNodeId());
+	}
+
+	private IEclipsePreferences getDefaultPreferenceStore() {
+		return DefaultScope.INSTANCE.getNode(JavaManipulation.getPreferenceNodeId());
+	}
+
+	private String getPreference(String key, String def) {
+		String str= getPreferenceStore().get(key, null);
+		if( str == null ) {
+			str= getDefaultPreferenceStore().get(key, null);
+		}
+		return str == null ? def : str;
 	}
 
 	private synchronized StringMatcher[] getStringMatchers() {
 		if (fStringMatchers == null) {
-			String str= PreferenceConstants.getPreferenceStore().getString(PreferenceConstants.TYPEFILTER_ENABLED);
+			String str= getPreference(PreferenceConstantsCore.TYPEFILTER_ENABLED, null);
```

Comment:
I believe this should pass "" as default instead of null as this is what happens in ScopedPreferenceStore for example (passes back at least "") and avoid a possible NPE.